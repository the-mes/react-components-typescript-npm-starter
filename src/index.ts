export { HelloWorld as default } from './components/HelloWorld';
