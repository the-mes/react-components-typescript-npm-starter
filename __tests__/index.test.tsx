import React from 'react';
import { render } from '@testing-library/react';

import { HelloWorld } from '../src';

describe('HelloWorld', () => {
  it('should render component with default text', () => {
    const {
      // debug,
      getByText,
      container,
    } = render(<HelloWorld />);

    // debug();

    expect(getByText('Hello World!')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render component with custom text', () => {
    const {
      // debug,
      getByText,
      container,
    } = render(<HelloWorld text="Lorem Ipsum" />);

    // debug();

    expect(getByText('Lorem Ipsum')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
