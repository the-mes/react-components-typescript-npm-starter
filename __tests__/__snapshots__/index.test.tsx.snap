// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld should render component with custom text 1`] = `
<div>
  <p>
    Lorem Ipsum
  </p>
</div>
`;

exports[`HelloWorld should render component with default text 1`] = `
<div>
  <p>
    Hello World!
  </p>
</div>
`;
